File: Interviewbit/linked-lists/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode A) {
        ListNode front=A,prev=A, next=A;
        while(next!=null){
            
            while(next!=null && prev.val==next.val){
                next=next.next;
            }
            prev.next=next;
            prev=next;
        }
        return front;
    }
}

File: Interviewbit/linked-lists/remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
ListNode* Solution::deleteDuplicates(ListNode* A) {
    
    ListNode *p=A,*q=A;
    
    while(q!=NULL){
        int value=q->val;
        while(q!=NULL&&q->val==value){
            q=q->next;
        }
        p->next=q;
        p=q;
    }
    return A;
}
